generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -8,42 +8,22 @@
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import java.util.stream.StreamSupport;
 
 public class StandardFieldValueSourceEvaluator implements FieldValueSourceEvaluator {
     private final MustContainRestrictionReducer mustContainRestrictionReducer = new MustContainRestrictionReducer();
 
     public Set<FieldValueSource> getFieldValueSources(FieldSpec fieldSpec){
-        Set<FieldValueSource> validSources = new HashSet<>();
-        MustContainRestriction mustContainRestriction = fieldSpec.getMustContainRestriction();
 
-        // check nullability...
-        if (determineNullabilityAndDecideWhetherToHalt(validSources, fieldSpec))
-            return validSources;
+        if (mustBeNull(fieldSpec)){
+            return Collections.singleton(nullOnlySource());
+        }
 
-        // if there's a whitelist, we can just output that
         if (fieldSpec.getSetRestrictions() != null && fieldSpec.getSetRestrictions().getWhitelist() != null) {
-            Set<Object> whitelist = fieldSpec.getSetRestrictions().getWhitelist();
-
-            // If we have values that must be included we need to check that those values are included in the whitelist
-            if (mustContainRestriction != null) {
-                whitelist = Stream.concat(whitelist.stream(),
-                getNotNullSetRestrictionFilterOnMustContainRestriction(mustContainRestriction)
-                    .flatMap(o -> o.getSetRestrictions().getWhitelist().stream())).collect(Collectors.toSet());
-            }
-
-            Stream<Object> validSourcesValues = validSources
-                .stream()
-                .flatMap(valueSource -> StreamSupport.stream(valueSource.generateAllValues().spliterator(), false));
-            Stream<Object> whitelistAndValidSourcesValues = Stream
-                .concat(whitelist.stream(), validSourcesValues);
-
-            return Collections.singleton(
-                new CannedValuesFieldValueSource(whitelistAndValidSourcesValues.collect(Collectors.toList()))
-            );
+            return getWhitelistSources(fieldSpec);
         }
 
-        if (mustContainRestriction != null) {
+        Set<FieldValueSource> validSources = new LinkedHashSet<>();
```

Comment:
Can we change this to a `List<>` and update the dependency graph to use `List<>` instead of `Set<>`